pkg/storegateway/bucket.go

```diff
@@ -2347,11 +2351,11 @@ func (r *bucketIndexReader) loadSeries(ctx context.Context, ids []storage.Series
 			level.Warn(r.block.logger).Log("msg", "series size exceeded expected size; refetching", "id", id, "series length", n+int(l), "maxSeriesSize", maxSeriesSize)
 
 			// Fetch plus to get the size of next one if exists.
-			return r.loadSeries(ctx, ids[i:], true, uint64(id), uint64(id)+uint64(n+int(l)+1))
+			return r.loadSeries(ctx, ids[i:], true, uint64(id), uint64(id)+uint64(n+int(l)+1), loaded, stats)
 		}
 		c = c[n : n+int(l)]
 		r.mtx.Lock()
-		r.loadedSeries[id] = c
+		loaded.series[id] = c
 		r.block.indexCache.StoreSeriesForRef(ctx, r.block.userID, r.block.meta.ULID, id, c)
 		r.mtx.Unlock()
```

Comment:
I wonder if we even need a mutex here. The index cache has its own mutex and we should only ever assign `loaded.series[id] = c` once for every possible `id`.

Reply:
Same question applies for the usage here

https://github.com/grafana/mimir/blob/6540e1270d17fc2dcbefe9b352c07a8533f39ab2/pkg/storegateway/bucket.go#L2175-L2181

if we don't need a mutex in either place we can just remove the field from `bucketIndexReader`

Reply:
sounds good, thank you